marlowe-actus/src/Language/Marlowe/ACTUS/Model/Utility/YearFraction.hs

```diff
@@ -40,6 +42,8 @@ yearFraction DCC_A_365 startDay endDay _
   = 0.0
 
 yearFraction DCC_E30_360ISDA startDay endDay maturityDate
+  | isNothing maturityDate
+  = error $ "DCC_E30_360ISDA requires maturity date"
```

Comment:
In this case, we could match directly in the clauses:
```haskell
yearFraction DCC_E30_360ISDA _ _ Nothing = error "DCC_E30_360ISDA requires maturity date"
yearFraction DCC_E30_360ISDA startDay endDay (Just maturityDate)
  | startDay <= endDay
  = let
  ...
```
And then we wouldn't need the `fromJust` in line 55